src/hpr_config_worker.erl

```diff
@@ -0,0 +1,170 @@
+%% ------------------------------------------------------------------
+-export([
+    init/1,
+    handle_continue/2,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    terminate/2
+]).
+
+-record(state, {
+    host :: string(),
+    port :: integer(),
+    connection :: grpc_client:connection() | undefined,
+    stream :: grpc_client:stream() | undefined,
+    dets_backup_path :: string() | undefined,
+    dets :: reference() | undefined
+}).
+
+-type config_worker_opts() :: #{
+    enabled := boolean(),
+    host := string(),
+    port := integer(),
+    dets_backup_path => string()
+}.
+
+-type routes_req_v1_map() :: #{
+    routes := [map()]
+}.
+
+-define(SERVER, ?MODULE).
+-define(RCV_CFG_UPDATE, receive_config_update).
+-define(RCV_TIMEOUT, timer:seconds(1)).
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+-spec start_link(config_worker_opts()) -> any().
+start_link(#{host := Host, port := Port} = Args) when is_list(Host) andalso is_number(Port) ->
+    gen_server:start_link({local, ?SERVER}, ?SERVER, Args, []);
+start_link(_) ->
+    ignore.
+
+%% ------------------------------------------------------------------
+%% gen_server Function Definitions
+%% ------------------------------------------------------------------
+
+init(#{host := Host, port := Port} = Args) ->
+    Path = maps:get(dets_backup_path, Args, undefined),
+    State = #state{
+        host = Host,
+        port = Port,
+        connection = undefined,
+        stream = undefined,
+        dets_backup_path = Path,
+        dets = undefined
+    },
+    lager:info("starting config worker ~s:~p dets=~s", [Host, Port, Path]),
+    {ok, maybe_init_dets(State), {continue, connect}}.
+
+handle_continue(connect, #state{host = Host, port = Port} = State) ->
+    case grpc_client:connect(tcp, Host, Port) of
+        {ok, Connection} ->
+            lager:info("connected"),
+            {noreply, State#state{connection = Connection}, {continue, init_stream}};
+        {error, _E} ->
+            lager:error("failed to connect ~p", [_E]),
+            timer:sleep(timer:seconds(1)),
+            {noreply, State, {continue, connect}}
+    end;
+handle_continue(
+    init_stream,
+    #state{
+        connection = Connection
+    } = State
+) ->
+    {ok, Stream} = grpc_client:new_stream(
+        Connection, 'helium.config.config_service', route_updates, client_config_pb
+    ),
+    %% Sending Route Request
+    RouteReq = #{},
+    ok = grpc_client:send(Stream, RouteReq),
+    lager:info("stream initialized"),
+    ok = cfg_update(),
+    {noreply, State#state{stream = Stream}}.
+
+handle_call(Msg, _From, State) ->
+    {stop, {unimplemented_call, Msg}, State}.
+
+handle_cast(Msg, State) ->
+    {stop, {unimplemented_cast, Msg}, State}.
+
+handle_info(?RCV_CFG_UPDATE, #state{connection = Connection, stream = Stream} = State) ->
```

Comment:
This could also be done with a `handle_continue`. Then we don't need to msg the process.

Another possibility is to refactor this as a `gen_statem` and `next_event`, since we're basically using the first arg to `handle_continue` as the state.